### PR TITLE
Consistent read query escaping

### PR DESCRIPTION
**Pull request by calebmer**
This is a simple bug fix. As I camel case my APIs PostgREST was generating bad queries which look like this:

```sql
SELECT
  "api"."posts".*,
  row_to_json(postTopics.*) AS "postTopic"
FROM
  "api"."posts"
LEFT OUTER JOIN (
  SELECT
    "api"."postTopics".*
  FROM
    "api"."postTopics"
) AS postTopics
ON
  "postTopics"."id" = "posts"."postTopicId";
```

Note how some of the `postTopics` are not quoted. This is because those values are not run through `pgFmtIdent`. This caused an error as `posttopics` and `"postTopics"` do not match. This is a quick PR to fix that.

### CODE REVIEW

**Comment by begriffs** [comment]
Good catch. Unquoted identifiers make me worried about sql injection too.

Can you add a test for this change?

**Comment by calebmer** [comment]
Sure. I didn't initially because it would involving adding a couple tables.

**Comment by calebmer** [comment]
Sorry for taking so long, I had other priorities, but the tests are added. It's a couple of weird corner cases that include the error I was hitting. We can expand that suite as we find more corner cases.

File: test/Feature/QuerySpec.hs [hunk 1]
```diff
@@ -382,3 +382,18 @@ spec struct pool = around (withApp cfgDefault struct pool) $ do
       it "returns proper json" $
         post "/rpc/sayhello" [json| { "name": "world" } |] `shouldRespondWith`
           [json| [{"sayhello":"Hello, world"}] |]
+
+  describe "weird requests" $ do
+    it "can query as normal" $ do
```
Comment (begriffs) [inline test/Feature/QuerySpec.hs]: This test passes with the code on master, is this what you would expect?

File: test/Feature/QuerySpec.hs [hunk 2]
```diff
@@ -382,3 +382,18 @@ spec struct pool = around (withApp cfgDefault struct pool) $ do
...
+      get "/Escap3e;" `shouldRespondWith`
+        [json| [{"so6meIdColumn":1},{"so6meIdColumn":2},{"so6meIdColumn":3},{"so6meIdColumn":4},{"so6meIdColumn":5}] |]
+      get "/ghostBusters" `shouldRespondWith`
+        [json| [{"escapeId":1},{"escapeId":3},{"escapeId":5}] |]
+
+    it "will embed a collection" $
```
Comment (begriffs) [inline test/Feature/QuerySpec.hs]: This test passes with the code on master.

**Comment by calebmer** [comment]
Yes for both tests. The last one is really the issue I was hitting, but I figured as long as I was writing tests I could write some safeguards for weird escaping.

**Comment by begriffs** [comment]
Can you rebase please? I think the changelog has a merge conflict because Ruslan's change also adds an "Unreleased" section. Next time I make a new release I'll add an empty unreleased section ahead of time so that the next two pull requests don't conflict.

**Comment by calebmer** [comment]
Done, sorry again for taking so long.
